system/cu/cu_main.c

```diff
@@ -446,7 +446,7 @@ int main(int argc, FAR char *argv[])
 
   while (!cu->force_exit)
     {
-      int ch = getc(stdin);
+      signed char ch = getc(stdin);
```

Comment:
This seems to be improper fix as `getc` return value is truncated.
We should stay with `int ch = getc(stdin);`, but replace `write(cu->outfd, &ch, 1);` with
```
char c = ch;
write(cu->outfd, &c, 1);
```
Or we can switch from `int outfd` to `FAR FILE *outfile` and use `fputc(ch, cu->outfile)` instead of `write(cu->outfd, ...);`

Reply:
The issue with current code should only be a problem for BE architectures